pkg/utils/utils.go

```diff
@@ -45,3 +47,51 @@ func GetVfPciDevFromMAC(mac string) (string, error) {
 	}
 	return dev, err
 }
+
+// Get RDMA device prefix and index. e.g for mlx5_3: prefix is mlx5 and index is 3
+// Note: the index is not related to the kernel RDMA device index
+func getRdmaDevNamePrefixIndex(rdmaDev string) (prefix string, idx uint64, err error) {
+	s := strings.Split(rdmaDev, `_`)
+	if len(s) != 2 {
+		return "", 0, fmt.Errorf("unexpeded RDMA device format: %s", rdmaDev)
+	}
+	prefix = s[0]
+	idx, err = strconv.ParseUint(s[1], 0, 32)
+	if err != nil {
+		err = fmt.Errorf("failed to parse RDMA device index: %s, %v", rdmaDev, err)
+	}
+	return prefix, idx, err
+}
+
+func getRdmaDevIndexFromName(rdmaDev string) (uint64, error) {
+	_, idx, err := getRdmaDevNamePrefixIndex(rdmaDev)
+	return idx, err
+}
+
+// Get the next RDMA device name for a given RDMA device prefix
+func GetNextRdmaDeviceName(prefix string, currDevs []string) (string, error) {
```

Comment:
@adrianchiris 
I think it is better to use PCI device based naming scheme from 
https://github.com/linux-rdma/rdma-core/blob/master/kernel-boot/rdma-persistent-naming.rules

And let rdma/hca operator to configure system for rdma device naming.